README.md

```diff
@@ -76,18 +116,207 @@ load(
 )
 ```
 
-Then use `codechecker_test()` rule passing targets you call CodeChecker for:
+Create a `codechecker_test()` target by passing other targets you'd like CodeChecker to analyze:
+
+<!-- TODO: Consider using https://github.com/bazelbuild/stardoc to document parameters -->
+```python
+codechecker_test(
+    name = "your_codechecker_rule_name",
+    targets = [
+        "your_target",
+    ],
+)
+```
+
+Then invoke bazel:
+
+```bash
+bazel build ://your_codechecker_rule_name
+```
+
+You can find the analysis results in the `bazel-bin/` folder, on which you
+can run [`CodeChecker store`](https://github.com/Ericsson/codechecker/blob/master/docs/web/user_guide.md#store)
+or [`CodeChecker parse`](https://github.com/Ericsson/codechecker/blob/master/docs/analyzer/user_guide.md#parse):
+
+```bash
+CodeChecker parse bazel-bin/path/to/results
+```
+
+### Build-only CodeChecker analysis: `codechecker()`
+
+This rule is functionally equivalent to `codechecker_test` but omits the test phase where either PASS or FAIL is printed.
+You can include and use it similarly as well:
+
+```python
+load(
+    "@bazel_codechecker//src:codechecker.bzl",
+    "codechecker"
+)
+```
+
+### Multiplatform CodeChechecker analysis: `codechecker_suite()`
+
+This rule is functionally equivalent to `codechecker_test` but allows for running on multiple platforms via the `platforms` parameter.
+You can include and use it similarly as well:
+
+```python
+load(
+    "@bazel_codechecker//src:codechecker.bzl",
+    "codechecker_suite"
+)
+```
+
+### `codechecker_config()`
+
+Using the Bazel rule `codechecker_config()` you can parse a CodeChecker [configuration file](https://github.com/Ericsson/codechecker/blob/master/docs/config_file.md).
+
+First, include the rule in your BUILD file:
+
+```python
+load(
+    "@bazel_codechecker//src:codechecker.bzl",
+    "codechecker_config"
+)
+```
+
+Create a CodeChecker configuration file e.g. `config.json` (see example [here](https://github.com/Szelethus/codechecker_bazel/blob/readme_update/test/config.json)) and parse it using `codechecker_config()`. 
+
+```python
+codechecker_config(
+    name = "your_codechecker_config",
+    config_file = ":config.json"
+)
+```
+
+Alternatively, you can assemble a CodeChecker configuration without a config file using the rule:
+
+```python
+codechecker_config(
+    name = "your_codechecker_config",
+    analyze = [
+        "--enable=bugprone-dangling-handle",
+        "--enable=bugprone-fold-init-type",
+        "--enable=misc-non-copyable-objects",
+        "--report-hash=context-free-v2",
+    ]
+)
+```
+
+You can now configure your `codechecker`, `codechecker_suite`, `codechecker_test` and `code_checker_test` targets using the above configuration:
 
 ```python
 codechecker_test(
     name = "your_codechecker_rule_name",
+    config = "your_codechecker_config",
+    targets = [
+        "your_target",
+    ],
+)
+```
+
+### Per-file CodeChecker analysis: `code_checker_test()`
```

Comment:
I actually consider per-file analysis as a drpo-in replacement for codechecker_test().
I had an idea to add a parameter, something like **per_file = true,** to make migration smoother.

Reply:
Yes, there is no point in keeping it separate for long. However, as long as we are in this demo stage, I'd still prefer to keep it documented before we merge/replace them.

Reply:
Created a new issue about this: https://github.com/Ericsson/codechecker_bazel/issues/31.